### PR TITLE
Upgrade tests.

### PR DESCRIPTION
**Pull request by jvanz**
Adds test script to validate the Helm chart upgrades.

## Test

```shell
make KUBEWARDEN_HELM_REPO_URL=http://jvanz.com/helm-charts/ KUBEWARDEN_DEFAULTS_CHART_OLD_VERSION=0.1.5 upgrade-test
```

### CODE REVIEW

File: tests/upgrade.bats [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+@test "[CRD upgrade] Upgrade CRDs" {
+	helm upgrade --wait --kube-context $CLUSTER_CONTEXT \
+		--namespace $NAMESPACE --create-namespace \
+		--version "1.0.0" \
+		kubewarden-crds $KUBEWARDEN_CHARTS_LOCATION/kubewarden-crds
+
+	helm upgrade --wait --namespace $NAMESPACE --kube-context $CLUSTER_CONTEXT \
+		--values $RESOURCES_DIR/default-kubewarden-controller-values.yaml \
+		--version "1.0.0" \
```
Comment (raulcabello) [inline tests/upgrade.bats:41]: same here

File: tests/upgrade.bats [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+@test "[CRD upgrade] Launch a privileged pod should fail" {
+	kubectl_apply_should_fail $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+}
+
+@test "[CRD upgrade] Upgrade CRDs" {
+	helm upgrade --wait --kube-context $CLUSTER_CONTEXT \
+		--namespace $NAMESPACE --create-namespace \
+		--version "1.0.0" \
```
[the remark this answers is not in the document]
Comment (jvanz) [reply tests/upgrade.bats:36]: Thanks! I've created the variable and forgot to update the files. :man_facepalming: 

Fixing this now!

File: Makefile [hunk 3]
```diff
@@ -46,15 +58,57 @@ bats = RESOURCES_DIR=$(RESOURCES_DIR) \
...
+endef
+
+define recreate-kubewarden-cluster
+	$(create-cluster)
+	$(install-cert-manager)
+	$(install-kubewarden)
+
+endef
+
+define generate-versioned-resources-dir
```
Comment (viccuad) [inline Makefile:99]: could it be possible to use the newly added `crdVersion` value in the chart? that simplifies things, and makes sure that value is tested.

Comment (jvanz) [reply Makefile:99]: I'm not sure if I understand exactly what you requested. But change the scripts to generated the versioned resources dir to decide which CRD version to use based on the values from the latest `kubewarden-defaults` released.

**Review by raulcabello — APPROVED** [review]
LGTM. thanks!

File: Makefile (hunk 3 above)
Comment (viccuad) [reply Makefile:104]: yep, looks good, thanks!

**Review by viccuad — APPROVED** [review]
LGTM!

**Comment by jvanz** [comment]
This PR now depends on https://github.com/kubewarden/helm-charts/pull/99. Moving to block until we merge it.

**Pull request merged**